src/converters.ts

```diff
@@ -211,55 +232,73 @@ export function transformCredentialInput(
   const credentialSubject = { ...input.credentialSubject, ...input.vc?.credentialSubject }
   if (!input.sub) {
     result.sub = input.credentialSubject?.id
-    delete credentialSubject.id
+    if (removeOriginalFields) {
+      delete credentialSubject.id
```

Comment:
If this `delete` would not be wrapped inside of the (unneeded, see comment above), then they could all be grouped at the end of the function. It's fine for now, since it cannot be implemented in a different way.

Reply:
right.. a bunch of them could be grouped more efficiently, but not all.
I preferred to keep them separate because it segments the code into sections that "deal with issuer", "deal with context", etc. which can later be implemented as distinct functions if the transforms grow too big.